### PR TITLE
l10n: improved interpolation

### PR DESCRIPTION
**Pull request by LeonardoRosaa**
Hey,

With this Pull Request, I have hope do resolved interpolation string `generate localizations` when there is a two placeholders to interpolate side by side.

Suppose have this `String`:

```json
"My name is {name}{lastname}"
```

Before change of this PR, `flutter_tool` for `generate localizations` generated it:

```json
"My name is $name${lastname}"
```

But when generate thus, the lint may to complain that isn't may did it. However, after this changes fixed it:
```json
"My name is $name$lastname"
```
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by christopherfujino** [comment]
@LeonardoRosaa I'm sorry I did not see this earlier. However, I see you checked ` I listed at least one issue that this PR fixes in the description above.`, however I do not see a listed issue? Is there one discussing the context of the bug?

**Comment by LeonardoRosaa** [comment]
> @LeonardoRosaa I'm sorry I did not see this earlier. However, I see you checked ` I listed at least one issue that this PR fixes in the description above.`, however I do not see a listed issue? Is there one discussing the context of the bug?

Sorry @christopherfujino, I did unchecked this option, because is not true. I did unconsciously.

**Comment by christopherfujino** [comment]
> > @LeonardoRosaa I'm sorry I did not see this earlier. However, I see you checked ` I listed at least one issue that this PR fixes in the description above.`, however I do not see a listed issue? Is there one discussing the context of the bug?
> 
> Sorry @christopherfujino, I did unchecked this option, because is not true. I did unconsciously.

Ok, could you add another unit test that exercises the simple example that you put in the description? I am not familiar enough with this code to verify this test; I see that you updated the existing unit test, however it is not obvious from the diff that the previous case was wrong and the current case is correct.

**Comment by LeonardoRosaa** [comment]
> > > @LeonardoRosaa I'm sorry I did not see this earlier. However, I see you checked ` I listed at least one issue that this PR fixes in the description above.`, however I do not see a listed issue? Is there one discussing the context of the bug?
> > 
> > 
> > Sorry @christopherfujino, I did unchecked this option, because is not true. I did unconsciously.
> 
> Ok, could you add another unit test that exercises the simple example that you put in the description? I am not familiar enough with this code to verify this test; I see that you updated the existing unit test, however it is not obvious from the diff that the previous case was wrong and the current case is correct.

Sure, I need some days for it :)

**Comment by christopherfujino** [comment]
> > > > @LeonardoRosaa I'm sorry I did not see this earlier. However, I see you checked ` I listed at least one issue that this PR fixes in the description above.`, however I do not see a listed issue? Is there one discussing the context of the bug?
> > > 
> > > 
> > > Sorry @christopherfujino, I did unchecked this option, because is not true. I did unconsciously.
> > 
> > 
> > Ok, could you add another unit test that exercises the simple example that you put in the description? I am not familiar enough with this code to verify this test; I see that you updated the existing unit test, however it is not obvious from the diff that the previous case was wrong and the current case is correct.
> 
> Sure, I need some days for it :)

Thanks! I know it's not fun, but I think it will help improve the maintainability of this code.

**Comment by LeonardoRosaa** [comment]
> > > > > @LeonardoRosaa I'm sorry I did not see this earlier. However, I see you checked ` I listed at least one issue that this PR fixes in the description above.`, however I do not see a listed issue? Is there one discussing the context of the bug?
> > > > 
> > > > 
> > > > Sorry @christopherfujino, I did unchecked this option, because is not true. I did unconsciously.
> > > 
> > > 
> > > Ok, could you add another unit test that exercises the simple example that you put in the description? I am not familiar enough with this code to verify this test; I see that you updated the existing unit test, however it is not obvious from the diff that the previous case was wrong and the current case is correct.
> > 
> > 
> > Sure, I need some days for it :)
> 
> Thanks! I know it's not fun, but I think it will help improve the maintainability of this code.

Don't worry, it's for better code.

**Comment by skia-gold** [comment]
Gold has detected about 9 new digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/92648

**Comment by christopherfujino** [comment]
@LeonardoRosaa thanks, the new test makes a lot of sense! I looked at some of your test failures and they look like they're all network related and not caused by your code. I will push an empty commit to your branch to retrigger them all...

**Comment by LeonardoRosaa** [comment]
> @LeonardoRosaa thanks, the new test makes a lot of sense! I looked at some of your test failures and they look like they're all network related and not caused by your code. I will push an empty commit to your branch to retrigger them all...

Thanks :)

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart [hunk 1]
```diff
@@ -388,9 +388,11 @@ bool _needsCurlyBracketStringInterpolation(String messageString, String placehol
     // Examples:
     // "'The number of {hours} elapsed is: 44'" // no curly brackets.
     // "'哈{hours}哈'" // no curly brackets.
+    // "'#placeholder#placeholder#'" // no curly brackets
     // "'m#hours#m'" // curly brackets.
     // "'I have to work _#hours#_' sometimes." // curly brackets.
-    final RegExp commonCaseRE = RegExp('[^a-zA-Z_][#{]$placeholder[#}][^a-zA-Z_]');
+    final RegExp commonCaseRE =
+        RegExp('([^a-zA-Z_]?(\$(.*){)?[#{]$placeholder[#}][^a-zA-Z_])');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/localizations/gen_l10n.dart:395]: I am worried about the new `(.*)`. Can you remind me what `.` accepts? What is the intention there?

**Review by christopherfujino — CHANGES_REQUESTED** [review]
I'm removing my approval pending @zanderso's question

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart (hunk 1 above)
Comment (LeonardoRosaa) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:395]: I don't remember exactly to why. But, it's matches all characters of placeholders.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:395]: I think that this is too general. Could you make a more specific regex pattern than `.*`?

**Comment by christopherfujino** [comment]
I'm closing this PR as stale. If you would like to address feedback in https://github.com/flutter/flutter/pull/92648#discussion_r824220870, feel free to re-open. Thanks!

**Pull request closed without merging**